### PR TITLE
tse simulator: (de)registerClientID

### PR DESCRIPTION
**Pull request by mhcp6**

### CODE REVIEW

File: stustapay/tse/diebold_nixdorf_usb/simulator.py [hunk 1]
```diff
@@ -278,6 +290,62 @@ def unblockuser(self, msg):
...
+        response = {}
+        # check if all Parameters are here
+        if "ClientID" not in msg or "Password" not in msg:
+            return dnerror(3)  # param missing
+        # check password_admin == 12345
+        try:
+            if base64.b64decode(msg["Password"]).decode("utf8") != self.password_admin:
+                self.password_block_counter += 1
+                return dnerror(30)  # password wrong
+        except:
```
Comment (TheJJ) [inline stustapay/tse/diebold_nixdorf_usb/simulator.py:331]: same here, should not be a catch-all, instead just the base64 decoding error

File: stustapay/tse/diebold_nixdorf_usb/simulator.py [hunk 2]
```diff
@@ -278,6 +290,62 @@ def unblockuser(self, msg):
...
+            return dnerror(9)  # base64
+
+        if len(msg["ClientID"]) > 30:
+            return dnerror(4)
+
+        if msg["ClientID"] in self.current_transactions:
+            if not self.current_transactions[msg["ClientID"]]:
+                del self.current_transactions[msg["ClientID"]]
+            else:
+                return dnerror(asdf)
```
Comment (TheJJ) [inline stustapay/tse/diebold_nixdorf_usb/simulator.py:341]: which dnerror is it? :)

**Review by TheJJ — CHANGES_REQUESTED** [review]

File: stustapay/tse/diebold_nixdorf_usb/simulator.py [hunk 3]
```diff
@@ -278,6 +290,62 @@ def unblockuser(self, msg):
...
+        response = {}
+        # check if all Parameters are here
+        if "ClientID" not in msg or "Password" not in msg:
+            return dnerror(3)  # param missing
+        # check password_admin == 12345
+        try:
+            if base64.b64decode(msg["Password"]).decode("utf8") != self.password_admin:
+                self.password_block_counter += 1
+                return dnerror(30)  # password wrong
+        except:
```
[the remark this answers is not in the document]
Comment (mhcp6) [reply stustapay/tse/diebold_nixdorf_usb/simulator.py]: now binascii.error

File: stustapay/tse/diebold_nixdorf_usb/simulator.py (hunk 1 above)
Comment (mhcp6) [reply stustapay/tse/diebold_nixdorf_usb/simulator.py]: now binascii.error

File: stustapay/tse/diebold_nixdorf_usb/simulator.py (hunk 2 above)
Comment (mhcp6) [reply stustapay/tse/diebold_nixdorf_usb/simulator.py]: 24

File: stustapay/tse/diebold_nixdorf_usb/simulator.py [hunk 4]
```diff
@@ -24,7 +24,8 @@ def __init__(self):
         self.puk_block_counter = 0
         self.transnr = 0
         self.signctr = 0
-        self.current_transactions = []
+        self.current_transactions = {}
+        self.current_transactions["POS001"] = set()
```
[the remark this answers is not in the document]
Comment (mhcp6) [reply stustapay/tse/diebold_nixdorf_usb/simulator.py:29]: for testing, so you don't have to register first

**Review by TheJJ — APPROVED** [review]

**Pull request merged**